modules/tinymce/src/themes/silver/main/ts/ui/throbber/Throbber.ts

```diff
@@ -62,19 +54,59 @@ const renderThrobber = (spec): AlloySpec => ({
     }
   },
   behaviours: Behaviour.derive([
-    Replacing.config({})
+    Replacing.config({}),
+    Blocking.config({
+      focus: false
+    }),
+    Composing.config({
+      find: (comp) => Arr.head(comp.components())
+    })
   ]),
   components: [ ]
 });
 
+const focusBusyComponent = (throbber: AlloyComponent): void => {
+  Composing.getCurrent(throbber).each((comp) => {
+    Keying.focusIn(comp);
+  });
+};
+
+/*
+* If the throbber has been toggled on, only focus the throbber if the editor had focus as we don't to steal focus if it is on an input or dialog
+* If the throbber has been toggled off, only put focus back on the editor if the throbber had focus.
+* The next logical focus transition from the throbber is to put it back on the editor
+*/
+const handleFocus = (editor: Editor, throbber: AlloyComponent, state: boolean, throbberFocus: boolean) => {
+  if (state) {
+    if (editor.hasFocus()) {
+      focusBusyComponent(throbber);
+    }
+  } else {
+    if (throbberFocus) {
+      editor.focus();
+    }
+  }
+};
+
 const setup = (editor: Editor, lazyThrobber: () => AlloyComponent, sharedBackstage: UiFactoryBackstageShared) => {
   const throbberState = Cell<boolean>(false);
   const timer = Cell<Optional<number>>(Optional.none());
 
+  // Make sure that when the editor is focused while the throbber is enabled, the focus is moved back to the throbber
+  // This covers native focus and editor.focus() invocations
+  editor.on('focus', () => {
+    if (throbberState.get()) {
+      focusBusyComponent(lazyThrobber());
+    }
+  });
+
   const toggle = (state: boolean) => {
     if (state !== throbberState.get()) {
-      toggleThrobber(lazyThrobber(), state, sharedBackstage.providers);
+      const throbber = lazyThrobber();
+      const throbberFocus = Composing.getCurrent(throbber).exists(Focusing.isFocused);
```

Comment:
This isn't used in all paths, so I think this would be cleaner if you only did the lookup inside `handleFocus` for the `state === false` path.

- [ ] Move this inside the `handleFocus` function

Reply:
Have moved the focus logic into the `toggleThrobber` function as it is simple enough now and for toggle off, I need check the throbber focus before it is removed.